com.unity.render-pipelines.universal/Runtime/LightCookieManager.cs

```diff
@@ -115,6 +120,97 @@ public void Resize(int size)
             }
         }
 
+        private struct ShaderBitArray
+        {
+            const int kBitsPerElement = 32;
+            const int kElementShift = 5;
+            const int kElementMask = (1 << kElementShift) - 1;
+
+            private float[] m_data;
+
+            public int elemLength => m_data == null ? 0 : m_data.Length;
+            public int bitCapacity => elemLength * kBitsPerElement;
+            public float[] data => m_data;
+
+            public void Resize(int bitCount)
+            {
+                if (bitCapacity > bitCount)
+                    return;
+
+                int newElemCount = ((bitCount + (kBitsPerElement - 1)) / kBitsPerElement);
+                if (newElemCount == m_data?.Length)
+                    return;
+
+                var newData = new float[newElemCount];
+                if (m_data != null)
+                {
+                    for (int i = 0; i < m_data.Length; i++)
+                        newData[i] = m_data[i];
+                }
+                m_data = newData;
+            }
+
+            public void Clear()
+            {
+                for (int i = 0; i < m_data.Length; i++)
+                    m_data[i] = 0;
```

Comment:
If that is changed to memcpy this could be changed to memset

Reply:
I decided to leave this as is, it avoids doing unsafe and fixed blocks.